src/util/OAuth2Util.js

```diff
@@ -95,6 +95,72 @@ define(['okta', './Enums', './Errors'], function (Okta, Enums, Errors) {
     }
   };
 
+  // Parse through the OAuth 'authParams' object to ensure the 'openid' scope is
+  // included (if required)
+  util.addOrRemoveOpenIdScope = function (authParams) {
+    if (!authParams.responseType) {
+      return;
+    }
+
+    //  Convert scope into an Array
+    var scope = Array.isArray(authParams.scopes) ? authParams.scopes : authParams.scopes.split(' ');
+
+    // Remove the 'openid' scope, as it is only required if an 'id_token' is requested
+    if (scope.includes('openid')) {
+      scope.splice(scope.indexOf('openid'), 1);
+    }
+
+    // Add the 'openid' scope
+    if (authParams.responseType.includes('id_token')) {
+      scope.push('openid');
+    }
+
+    return scope;
+  };
+
+  // Utility handlers for mapping convenience keys to OAuth params
+  util.getResponseType = function (options) {
+    var responseType = [];
+    if (options.getIdToken !== false) {
+      responseType.push('id_token');
+    }
+
+    if (options.getAccessToken) {
+      responseType.push('token');
+    }
+
+    return responseType;
+  };
+
+  util.transformShowSignInToGetTokensOptions = function(options, config = {}) {
+    // Override specific OAuth/OIDC values
+    var renderOptions = {
+      clientId: options.clientId,
+      redirectUri: options.redirectUri,
+      authParams: {
+        display: 'page',
+        responseMode: 'fragment',
+        responseType: util.getResponseType(options),
+        scopes: options.scope || (config.authParams && config.authParams.scopes) || ['openid']
+      }
+    };
+
+    if (options.authorizationServerId) {
+      // Map the authorizationServerId to issuer
+      renderOptions.authParams.issuer = options.authorizationServerId;
+    }
+
+    // Ensure the 'openid' scope is provided when an 'id_token' is requested.
+    // If the 'openid' scope is present and isn't needed, remove it.
+    renderOptions.authParams.scopes = util.addOrRemoveOpenIdScope(renderOptions.authParams);
+
+    // Remove 'undefined' values
+    Object.keys(renderOptions).forEach(function(key) {
+      return renderOptions[key] === undefined && delete renderOptions[key];
+    });
```

Comment:
Per offline conversation: if it isn't necessary to remove the undefined properties, I don't think this should be here because it makes me wonder why they need to be removed, and I would expect code comments that explain why.